File: src/main/java/io/github/cadiboo/examplemod/server/ServerProxy.java
package io.github.cadiboo.examplemod.server;

import io.github.cadiboo.examplemod.ExampleMod;
import io.github.cadiboo.examplemod.util.IProxy;
import net.minecraft.util.text.translation.I18n;
import net.minecraftforge.fml.relauncher.Side;

/**
 * The version of IProxy that gets injected into {@link ExampleMod#proxy} on a PHYSICAL/DEDICATED SERVER
 */
public class ServerProxy implements IProxy {

	@Override
	public String localize(final String unlocalized) {
		return I18n.translateToLocal(unlocalized);
	}

	@Override
	public String localizeAndFormat(final String unlocalized, final Object... args) {
		return I18n.translateToLocalFormatted(unlocalized, args);
	}

	@Override
	public Side getPhysicalSide() {
		return Side.SERVER;
	}

}

File: src/main/kotlin/io/github/cadiboo/examplemod/server/ServerProxy.kt
package io.github.cadiboo.examplemod.server

import io.github.cadiboo.examplemod.ExampleMod
import io.github.cadiboo.examplemod.util.IProxy
import net.minecraft.util.text.translation.I18n
import static net.minecraftforge.fml.relauncher.Side.SERVER

/**
 * The version of IProxy that gets injected into [ExampleMod.proxy] on a PHYSICAL/DEDICATED SERVER
 */
class ServerProxy : IProxy {

    override fun  getPhysicalSide() {
        return SERVER
    }

    override fun localize(unlocalized: String): String {
        return I18n.translateToLocal(unlocalized)
    }

    override fun localizeAndFormat(unlocalized: String, vararg args: Object): String {
        return I18n.translateToLocalFormatted(unlocalized, args)
    }

}
